iodata/formats/qchemlog.py

```diff
@@ -0,0 +1,368 @@
+                     'run_type', 'athessian', 'lot', 'obasis_name']
+    result = {label: data[label] for label in result_labels if data.get(label) is not None}
+    # add number of electrons
+    result['nelec'] = data['alpha_elec'] + data['beta_elec']
+
+    # mulliken charges
+    if data.get("mulliken_charges") is not None:
+        result["atcharges"] = {"mulliken": data["mulliken_charges"]}
+
+    # build molecular orbitals
+    # ------------------------
+    # restricted case
+    if not data['unrestricted']:
+        mo_energies = np.concatenate((data['mo_a_occ'], data['mo_a_vir']), axis=0)
+        mo_coeffs = np.empty((data['nbasis'], data['norba'])) * np.nan
+        mo_occs = np.zeros(mo_coeffs.shape[1])
+        mo_occs[:data['alpha_elec']] = 1.0
+        mo_occs[:data['beta_elec']] += 1.0
+        mo = MolecularOrbitals("restricted", data['norba'], data['norba'],
+                               mo_occs, mo_coeffs, mo_energies, None)
+    # unrestricted case
+    else:
+        mo_energies = np.concatenate((data['mo_a_occ'], data['mo_a_vir'],
+                                      data['mo_b_occ'], data['mo_b_vir']), axis=0)
+        mo_coeffs = np.empty((data['nbasis'], data['norba'] + data['norbb'])) * np.nan
+        mo_occs = np.zeros(mo_coeffs.shape[1])
+        # number of alpha & beta electrons and number of alpha molecular orbitals
+        na, nb = data['alpha_elec'], data['beta_elec']
+        na_mo = len(data['mo_a_occ']) + len(data['mo_a_vir'])
+        mo_occs[:na] = 1.0
+        mo_occs[na_mo: na_mo + nb] = 1.0
+        mo = MolecularOrbitals("unrestricted", data['norba'], data['norbb'],
+                               mo_occs, mo_coeffs, mo_energies, None)
+    result['mo'] = mo
+
+    # moments
+    moments = {}
+    if 'dipole' in data:
+        moments[(1, 'c')] = data['dipole']
+    if 'quadrupole' in data:
+        # Convert to alphabetical ordering: xx, xy, xz, yy, yz, zz
+        moments[(2, 'c')] = data['quadrupole'][[0, 1, 3, 2, 4, 5]]
+    if moments:
+        result['moments'] = moments
+
+    # extra dictionary
+    # ----------------
+    # add labels to extra dictionary if they are loaded
+    extra_labels = ['spin_multi', 'nuclear_repulsion_energy',
+                    'polarizability_tensor', 'imaginary_freq', 'vib_energy']
+    extra = {label: data[label] for label in extra_labels if data.get(label) is not None}
+    # if present, convert vibrational energy from kcal/mol to "atomic units + K"
+    if 'vib_energy' in extra:
+        extra['vib_energy'] *= kcalmol
+
+    # if present, convert enthalpy terms from kcal/mol to "atomic units + K"
+    if 'enthalpy_dict' in data:
+        extra['enthalpy_dict'] = {k: v * kcalmol for k, v in data['enthalpy_dict'].items()}
+
+    # if present, convert entropy terms from cal/mol.K to "atomic units + Kalvin"
+    if 'entropy_dict' in data:
+        extra['entropy_dict'] = {k: v * calmol for k, v in data['entropy_dict'].items()}
```

Comment:
@tovrstra what are your thoughts on storing thermodynamic properties cleanly? I am worried that `extra` is becoming a place to dump things that are not clearly defined... Here, it makes sense to probably parse temperature and pressure as well, but then you add more items to the dictionary.

@wilhadams is there any specific convention for storing thermodynamic properties in Schema?

Reply:
@FarnazH currently there's no convention.

Reply:
I would not mind adding these to the extra dictionary for now. It might be useful to document that the keys in extra are never to be considered API stable. This gives us the freedom to migrate some commonly used items in `extra` to more final attributes of `IOData`. @FarnazH: would that make sense?

Reply:
Make sense @tovrstra. I will update the `extra` docstring in a future PR.